src/test/scala/com/datastax/gatling/plugin/simulations/cql/BoundCqlTypesSimulation.scala

```diff
@@ -110,37 +116,43 @@ class BoundCqlTypesSimulation extends BaseCqlSimulation {
   )
   // End counter details
 
+  // A predicate function can be useful when we want to do multiple comparisons on data in a single row
+  private def preparedCqlPredicate(row:Row):Boolean =
+    row.getBoolean("boolean_type") && row.isNull("null_type")
+
+  // In most cases we can simply extract a value from the row and compare that extracted value to an expected value
+  // via the Gatling API
+  private def counterCqlExtract(row:Row):Long =
+    row.getLong("counter_type")
 
   val scn = scenario("BoundCqlStatement")
 
       .feed(preparedFeed)
       .exec(insertPreparedCql
-          .check(exhausted is true)
-          .check(rowCount is 0) // "normal" INSERTs don't return anything
+          .check(resultSet.transform(_.hasMorePages) is false)
+          .check(resultSet.transform(_.remaining) is 0) // "normal" INSERTs don't return anything
       )
       .pause(100.millis)
 
       .exec(selectPreparedCql
           .withParams(List("uuid_type"))
 
-          .check(rowCount is 1)
-          .check(columnValue("name") not "")
-          .check(columnValue("null_type") not "test")
-          .check(columnValue("boolean_type") is true)
+          .check(resultSet.transform(_.remaining) is 1)
+          .check(resultSet.transform(rs => preparedCqlPredicate(rs.one)) is true)
```

Comment:
We want to do multiple checks here for every Row, but note that the Row accessor off of an AsyncResultSet is one(), which has the side effect of advancing the current row on each call.  So we need to implement a discrete function to perform multiple checks on a single row... thus preparedCqPredicate().